### ISSUE TITLE
Weights not updating after manually setting

### ISSUE THREAD

**Issue by nassimmokhtari** [keras-team/keras#16539]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):  2.6.0
- Python version: 3.9.7
- GPU model and memory:  NVIDIA RTX A3000 Laptop GPU 6Go

**Describe the problem**.
After setting manually the weights of a specific layer, the model is not able to learn: the bias still getting updated, but not the kernel weights.

**Source code / logs**.
Here is an example where I am setting kernel weights to 0.75 and bias to 0.25 for the first layer, using the code bellow :
```
from keras.layers import Dense
from keras.models import Sequential

model = Sequential()
model.add(Dense(10,input_shape=(784,)))
model.add(Dense(10))
model.compile(loss='categorical_crossentropy',metrics=['accuracy'])
old_weights,old_bias = model.layers[0].get_weights()
shape_w = old_weights.shape
initial_weights=np.ones(shape_w)*0.75
shape_b = old_bias.shape
initial_bias = np.ones(shape_b)*0.25
model.layers[0].set_weights([initial_weights,initial_bias])
model.fit(X_train,Y_train,epochs=10)
```
But when I print the weights after training I am getting : 

```
[array([[0.75, 0.75, 0.75, ..., 0.75, 0.75, 0.75],
       [0.75, 0.75, 0.75, ..., 0.75, 0.75, 0.75],
       [0.75, 0.75, 0.75, ..., 0.75, 0.75, 0.75],
       ...,
       [0.75, 0.75, 0.75, ..., 0.75, 0.75, 0.75],
       [0.75, 0.75, 0.75, ..., 0.75, 0.75, 0.75],
       [0.75, 0.75, 0.75, ..., 0.75, 0.75, 0.75]], dtype=float32), array([0.24182914, 0.26478815, 0.2445423 , 0.25399897, 0.24518993,
       0.24241278, 0.23505618, 0.24735092, 0.2545016 , 0.24763964],
      dtype=float32)]
```

**Comment by sushreebarsa** [comment]
@nassimmokhtari  I tried to  replicate the issue on colab and faced a different error. Could you  please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/52c100720d7419c92776dec0bebde6e9/16539.ipynb)  and confirm the same? Thank you!

**Comment by nassimmokhtari** [comment]
@sushreebarsa The error you are facing is due to the missing dataset (X_train and Y_train), I'm using the MINST dataset, where I flatten each image. You can try using MINST or any other dataset (just be sure that the input shape are matching). Thanks

**Comment by sushreebarsa** [comment]
@nassimmokhtari Could you please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/a13746cd42591ecb46a038c5b57b6521/keras16539.ipynb) and confirm the issue reported ?
Thank you!

**Comment by nassimmokhtari** [comment]
@sushreebarsa This issue is due to the output shape, the network is using one hot codification for the output, and the label you provide are represented as an Integer. You can find [here](https://drive.google.com/file/d/1tzfjGst3CGVRq_mukK_CERJ1Sge57-Gn/view?usp=sharing) the dataset I'm using, the dataset is already processed, inputs are flattened and labels are encoded as one hot. Thanks

**Comment by sushreebarsa** [comment]
@gowthamkpr I was able to replicate the issue on colab , please find the [gist](https://colab.research.google.com/gist/sushreebarsa/6c6a08b7976d4bcfb87066df846ec392/kerasgist16539.ipynb) here.
Thank you!

**Comment by qlzh727** [comment]
The core issue here is that the model weights are symmetric matrix, esp for dense layer, this means each of the neuron is contributing same weights to the output, and this will cause they get updates at back propagation.

Also you model is ill formatted since it doesn't have any activation in the last dense layer. If you change to have activation='sigmoid' for last dense layer, and inspect the weights before/after the training, you should be able to see the change.
